fvm/transactionInvoker.go

```diff
@@ -308,7 +290,16 @@ func valueDeclarations(ctx *Context, env *TransactionEnv) []runtime.ValueDeclara
 						panic(errors.NewValueErrorf(invocation.Arguments[0].String(),
 							"second argument of setAccountFrozen must be a boolean"))
 					}
-					err := env.SetAccountFrozen(common.Address(address), bool(frozen))
+					var err error
+					switch env := env.(type) {
+					case *TransactionEnv:
+						err = env.SetAccountFrozen(common.Address(address), bool(frozen))
+					case *ScriptEnv:
+						err = env.SetAccountFrozen(common.Address(address), bool(frozen))
+					default:
+						err = errors.NewOperationNotSupportedError("SetAccountFrozen")
+					}
```

Comment:
not sure why we do this ? isn't suppose tobe the env returning the notsupportedError itself?

Reply:
The problem is that the `Environment` interface does not have the `SetAccountFrozen` method on it, and I did not want to add it, as it is only relevant on the `TransactionEnv`. 

I agree It looks weird...

I changed around a bit. I removed the `SetAccountFrozen` from the `ScriptEnv` altogether, as it was only used here, and shortened this part: https://github.com/onflow/flow-go/pull/1318/commits/b659c5e0b38305f25ed5c01980c34eb578152a5f